embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/fakes/FakePersistableEmbraceSpan.kt

```diff
@@ -4,25 +4,34 @@ import io.embrace.android.embracesdk.arch.destination.SpanEventData
 import io.embrace.android.embracesdk.arch.schema.EmbType
 import io.embrace.android.embracesdk.arch.schema.SchemaType
 import io.embrace.android.embracesdk.arch.schema.TelemetryType
+import io.embrace.android.embracesdk.internal.clock.millisToNanos
+import io.embrace.android.embracesdk.internal.clock.normalizeTimestampAsMillis
 import io.embrace.android.embracesdk.internal.payload.Span
+import io.embrace.android.embracesdk.internal.payload.toNewPayload
 import io.embrace.android.embracesdk.spans.EmbraceSpan
+import io.embrace.android.embracesdk.spans.EmbraceSpanEvent
 import io.embrace.android.embracesdk.spans.ErrorCode
 import io.embrace.android.embracesdk.spans.PersistableEmbraceSpan
 import io.opentelemetry.sdk.trace.IdGenerator
+import java.util.concurrent.ConcurrentLinkedQueue
 
 internal class FakePersistableEmbraceSpan(
     override val parent: EmbraceSpan?,
     val name: String? = null,
     val type: TelemetryType = EmbType.Performance.Default,
-    val internal: Boolean = true
+    val internal: Boolean = true,
+    private val fakeClock: FakeClock = FakeClock()
 ) : PersistableEmbraceSpan {
 
     val attributes = mutableMapOf<String, String>()
-    val events = mutableListOf<SpanEventData>()
+    val events = ConcurrentLinkedQueue<EmbraceSpanEvent>()
```

Comment:
This wasn't used for anything at the tip of master - but I added support to add a custom breadcrumb to it as a span event just in case it's being used to test the session span in some uncommitted PR.